tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+    }).store,
+    ...props
+  } = {}) {
+    return shallowUntilTarget(
+      <UserProfileEdit i18n={i18n} params={params} store={store} {...props} />,
+      UserProfileEditBase,
+      { shallowOptions: { context: { i18n } } }
+    );
+  }
+
+  function mountUserProfileEdit({
+    errorHandler = createStubErrorHandler(),
+    i18n = fakeI18n(),
+    params = { username: 'tofumatt' },
+    store = dispatchSignInActions({
+      userProps: defaultUserProps(),
+    }).store,
+    ...props
+  } = {}) {
+    return mount(
+      <Provider store={store}>
+        <UserProfileEdit
+          errorHandler={errorHandler}
+          i18n={i18n}
+          params={params}
+          store={store}
+          {...props}
+        />
+      </Provider>,
+      { context: { i18n } }
+    );
+  }
+
+  it('renders user profile page', () => {
+    const root = renderUserProfileEdit();
+
+    expect(root.find('.UserProfileEdit')).toHaveLength(1);
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const username = 'i-am-not-tofumatt';
+
+    const root = renderUserProfileEdit({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount action if there is no username', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    // This happens when loading the user edit profile page of the current
+    // logged user (e.g., page refresh).
+    renderUserProfileEdit({ params: {}, store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('dispatches fetchUserAccount action if username param changes', () => {
+    const { params, store } = syncPropsAndParamsUsername('black-panther');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    root.setProps({ params: { username: 'killmonger' } });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const { params, store } = syncPropsAndParamsUsername('black-panther');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    root.setProps({ params });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders a username input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-username')).toHaveLength(1);
```

Comment:
Would it be good to check the value of this field as well?

Reply:
Indeed!